modules/education/src/main/java/edu/stanford/spezi/modules/education/videos/component/ExpandableSection.kt

```diff
@@ -0,0 +1,196 @@
+import edu.stanford.spezi.modules.education.videos.Video
+import edu.stanford.spezi.modules.education.videos.VideoItem
+
+@Composable
+fun SectionHeader(
+    text: String?,
+    isExpanded: Boolean,
+    onHeaderClicked: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable { onHeaderClicked() }
+            .padding(Spacings.medium),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        text?.let {
+            Text(
+                text = it,
+                style = titleLarge
+            )
+        }
+
+        Icon(
+            imageVector = getExpandIcon(isExpanded),
+            contentDescription = if (isExpanded) "Collapse" else "Expand"
+        )
+    }
+}
+
+@Composable
+fun ExpandableSection(
+    modifier: Modifier = Modifier,
+    title: String?,
+    description: String?,
+    videos: List<Video> = emptyList(),
+    expandedStartValue: Boolean = false,
+    onExpand: () -> Unit = {},
+    onActionClick: (Video) -> Unit = { _ -> },
+) {
+    var expanded by remember { mutableStateOf(expandedStartValue) }
+
+    ElevatedCard(
+        elevation = CardDefaults.cardElevation(defaultElevation = Elevations.Card.medium),
+        shape = RoundedCornerShape(Shapes.RoundedCorners.large),
+        colors = CardDefaults.cardColors(
+            containerColor = lightenColor(Colors.surface),
+        ),
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(Spacings.small)
+            .clickable { expanded = !expanded },
+    ) {
+        Column(
+            modifier = Modifier.background(lightenColor(Colors.surface))
+        ) {
+            SectionHeader(
+                text = title,
+                isExpanded = expanded,
+                onHeaderClicked = {
+                    expanded = !expanded
+                    onExpand()
+                }
+            )
+
+            description?.let {
+                Text(
+                    style = TextStyles.bodyMedium,
+                    text = it,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(Spacings.medium),
+                    color = MaterialTheme.colorScheme.onBackground,
+                )
+            }
+            AnimatedVisibility(
+                visible = expanded,
+                enter = expandVertically(),
+                exit = shrinkVertically()
+            ) {
+                Column {
+                    videos.forEach { video ->
+                        VideoItem(video,
+                            onVideoClick = {
+                                onActionClick(
+                                    video
+                                )
+                            })
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun getExpandIcon(expanded: Boolean): ImageVector {
```

Comment:
I would get rid of the return value here and use:

```kotlin
@Composable
fun ExpandIcon(expanded: Boolean) {
    val vector = if (expanded) Icons.Filled.KeyboardArrowUp else Icons.Filled.KeyboardArrowDown
    Icon(
        imageVector = vector,
        contentDescription = null,
    )
}
```

Reply:
Nice suggestion, thanks!